pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -58,6 +58,13 @@ class PublishOptions(object):
     :meth:`~pubtools.pulplib.Distributor.is_rsync` is ``True``.
     """
 
+    rsync_extra_args = pulp_attrib(default=None, type=frozenlist)
```

Comment:
Setting `type=frozenlist` seems to be the only way to make the list immutable and permit `default=None`.

Reply:
This won't make the list immutable because type doesn't really do anything, it's basically just for documentation purposes.  It doesn't actually do a conversion.

The problem is that my suggestion of converter=frozenlist was not correct if we want to allow None values.  Looking at other similar cases in the classes in this library, we use frozenlist_or_none_converter.

For example ModulemdUnit.artifacts is defined like this:

```python
    artifacts = pulp_attrib(
        default=None,
        type=list,
        converter=frozenlist_or_none_converter,
        pulp_field="artifacts",
    )
```

So we want the same thing here except we don't need `pulp_field`.